apis/refs/v1beta1/computerefs.go

```diff
@@ -27,24 +26,15 @@ import (
 )
 
 type ComputeNetworkRef struct {
-	/* The compute network selflink of form "projects/<project>/global/networks/<network>", when not managed by KCC. */
+	/* An external value of a `ComputeNetwork` resource, when not managed by KCC. */
```

Comment:
The idea is that the user should always be able to use `projects/<project>/global/networks/<network>`.  Our controllers will convert to the correct format - full URL or just the name etc.

I think @gemmahou is proposing in this PR that we allow users to specify other values, and when they do we pass them through to the underlying APIs as-is, so that if users used a different format with an existing resource, we don't break them.  That makes me think, maybe we should have a "strict" mode of parsing, and use strict for new resources?  cc @yuwenma

Reply:
There are rare cases when it is very difficult to achieve this consistent interface. In [this reference field for a ComputeNetwork](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/v1.121.0/config/servicemappings/servicedirectory.yaml#L53), the only accepted format is `projects/<projectNumber>/locations/global/networks/<networkID>`. Parsing project ID to project number requires specific permission that the KCC GSA may not have. So I'm inclined to keep accepting any form of the external value.

Reply:
+1 to Justin. We shall have a strict formatting rule for new Direct resources. 

For existing Beta resources, we have to do special handlings when migrating to Direct controller, but it should be isolated within that resource code base. Regarding the example @maqiuyujoyce mentioned, the direct resource shall not use the shared `computeRef`, but defines its own one which still accepting projectNumber for backward compatibility reasons.

Reply:
I think keeping a consistent format for most cases but still allow edge cases works. Regarding the example I mentioned above, I agree that we should support it separately, though I believe the purpose is not only for backwards compatibility, but is to avoid unnecessary permission issues. 

Here are the edge cases I think worth the effort to specialize:
1. Forming the actual value passed to the API requires an additional API call.
2. The value API accepts is not resource ID/self link/resource name related (e.g. the actual address managed by a ComputeAddress resource).

Though we want to add enough instructions for contributors to be able to make a decision themselves during the implementation.

@gemmahou I saw you closed the PR. Let me know if you want me to follow up somewhere else.